e2e/cypress/integration/plugins/upgrade_spec.js

```diff
@@ -0,0 +1,113 @@
+        cy.apiInitSetup().then(({team}) => {
+            cy.visit(`/${team.name}/channels/town-square`);
+
+            // #If Demo plugin is already enabled , uninstall it
+            cy.apiRemovePluginById(pluginIdDemo);
+            cy.visit('/admin_console/plugins/plugin_management');
+            cy.get('.admin-console__header', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').and('have.text', 'Plugin Management');
+        });
+    });
+
+    it('MT40 Plugin remains enabled when upgraded', () => {
+    // * Upload Demo plugin from the browser
+        const fileName1 = 'com.mattermost.demo-plugin-0.1.0.tar.gz';
+        const fileName2 = 'com.mattermost.demo-plugin-0.2.0.tar.gz';
+        const mimeType = 'application/gzip';
+        cy.fixture(fileName1, 'binary').
+            then(Cypress.Blob.binaryStringToBlob).
+            then((fileContent) => {
+                cy.get('input[type=file]').attachFile({fileContent, fileName: fileName1, mimeType});
+            });
+
+        cy.get('#uploadPlugin').scrollIntoView().should('be.visible').click().wait(TIMEOUTS.HALF_SEC);
+
+        // * Verify that the button shows correct text while uploading
+        cy.findByText('Uploading...', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible');
+
+        // * Verify that the button shows correct text and is disabled after upload
+        cy.findByText('Upload', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible');
+        cy.get('#uploadPlugin').and('be.disabled');
+
+        // # Enable demo plugin
+        doTaskOnPlugin(pluginIdDemo, () => {
+        // * Verify Demo Plugin title is shown
+            cy.waitUntil(() => cy.get('strong').scrollIntoView().should('be.visible').then((title) => {
+                return title[0].innerText === 'Demo Plugin';
+            }));
+
+            // # Click on Enable link
+            cy.findByText('Enable').click();
+        });
+
+        // * Verify V0.1.0 of plugin
+        cy.findByText(/0.1.0/).scrollIntoView().should('be.visible');
+
+        cy.get('#uploadPlugin').scrollIntoView().should('be.visible');
+
+        // # Upgrade plugin
+        cy.fixture(fileName2, 'binary').
+            then(Cypress.Blob.binaryStringToBlob).
+            then((fileContent) => {
+                cy.get('input[type=file]').attachFile({fileContent, fileName: fileName1, mimeType});
+            });
+
+        // * Verify that the button shows correct text while uploading
+        cy.get('#uploadPlugin').should('be.visible').click().wait(TIMEOUTS.HALF_SEC);
+
+        // # Confirm overwrite of plugin with same name
+        cy.get('#confirmModalButton').should('be.visible').click();
+
+        doTaskOnPlugin(pluginIdDemo, () => {
+        // * Verify plugin is running
+            waitForAlert('This plugin is running.');
+        });
+
+        // * Verify v0.2.0 of plugin
+        cy.findByText(/0.2.0/).scrollIntoView().should('be.visible');
+
+        cy.apiRemovePluginById(pluginIdDemo);
```

Comment:
I missed to mention to put it in the after hook so it's more likely to execute in case something failed or error occurred in the middle,
```javascript
after(() => {
    cy.apiRemovePluginById(pluginIdDemo);
});
```